crates/sui-graphql-rpc/src/types/move_package.rs

```diff
@@ -711,26 +716,26 @@ impl MovePackage {
                 async move {
                     let mut q = query!(
                         r#"
-                    SELECT
-                            p.original_id,
-                            o.*
-                    FROM
-                            packages p
-                    INNER JOIN
-                            objects_history o
-                    ON
-                            p.package_id = o.object_id
-                    AND     p.package_version = o.object_version
-                    AND     p.checkpoint_sequence_number = o.checkpoint_sequence_number
-                "#
+                            SELECT
```

Comment:
Is it possible to use built-in types instead of string queries?
That would make it easier to deal with schema changes and type discovery

Reply:
Unfortunately not -- and the reason is a little roundabout:

We have two frameworks for handling pagination in GraphQL -- one uses the typed diesel ORM, and the other uses these raw queries:

https://github.com/MystenLabs/sui/blob/274f340b6f830f8f2480f3500bb6b1b796b0fa02/crates/sui-graphql-rpc/src/types/cursor.rs#L59-L98

The former requires that the data being fetched comes from one `QuerySource` -- practically this is a single table -- and the latter does not have this restriction. These queries fit outside this mould because:

- They fetch from multiple tables,
- The system package query fetches from a different set of tables to the user package query.

It's possible that someone with better generics-fu than me could get diesel to submit to their will, but I couldn't manage it. I agree it's not ideal from a perspective of, e.g. using LSP to find all usages of the table. I resort to simple `grep` for that (and I've done that for `objects_history` now).